XamlAnimatedGif/Animator.cs

```diff
@@ -295,7 +337,25 @@ private async Task RenderFrameAsync(int frameIndex, CancellationToken cancellati
             var frame = _metadata.Frames[frameIndex];
             var desc = frame.Descriptor;
             var rect = GetFixedUpFrameRect(desc);
-            using var indexStream = await GetIndexStreamAsync(frame, cancellationToken);
+            var rows = desc.Interlace
+                ? InterlacedRows(rect.Height)
+                : NormalRows(rect.Height);
+
+            Stream indexStream = null;
+            if (!_cacheFrameDataInMemory)
+            {
+                indexStream = await GetIndexStreamAsync(frame, cancellationToken);
+            }
+
+            byte[][] indexDecompressedBytes = null;
+            if (_cacheFrameDataInMemory)
+            {
+                if (!_cachedFrameBytes.TryGetValue(frameIndex, out indexDecompressedBytes))
+                {
+                    return;
```

Comment:
Shouldn't this be an error? We should always have data for each frame, if we don't it's clearly a bug.

Reply:
yeah this check isn't needed so I removed it